nomad/server.go

```diff
@@ -1284,6 +1284,16 @@ func (s *Server) setupRaft() error {
 			return err
 		}
 
+		// Check Raft version and update the version file.
+		raftVersionFilePath := filepath.Join(path, "version")
+		raftVersionFileContent := fmt.Sprintf("%d", s.config.RaftConfig.ProtocolVersion)
+		if err := s.checkRaftVersionFile(raftVersionFilePath); err != nil {
+			return err
+		}
+		if err := ioutil.WriteFile(raftVersionFilePath, []byte(raftVersionFileContent), 0755); err != nil {
```

Comment:
I can't think of any reason we'd want this to be executable:

```suggestion
		if err := ioutil.WriteFile(raftVersionFilePath, []byte(raftVersionFileContent), 0644); err != nil {
```

Reply:
Ah thanks, I think it was bad copy-pasta from the [`peers.info` file](https://github.com/hashicorp/nomad/blob/main/nomad/server.go#L1333), which doesn't seem like it's supposed to be an executable either.

Reply:
I'll open a PR for that